dapr_agents/agents/durableagent/agent.py

```diff
@@ -843,14 +886,52 @@ def _construct_messages_with_instance_history(
             )
 
         # Get instance-specific chat history instead of global memory
+        if self.state is None:
+            logger.warning(
+                f"Agent state is None for instance {instance_id}, initializing empty state"
+            )
+            self.state = {}
+
         instance_data = self.state.get("instances", {}).get(instance_id)
         if instance_data is not None:
             instance_messages = instance_data.get("messages", [])
         else:
             instance_messages = []
 
-        # Convert instance messages to the format expected by prompt template
+        # Always include long-term memory (chat_history) for context
+        # This ensures agents have access to broadcast messages and persistent context
+        long_term_memory_data = self.state.get("chat_history", [])
+
+        # Convert long-term memory to dict format for LLM consumption
+        long_term_memory_messages = []
+        for msg in long_term_memory_data:
+            if isinstance(msg, dict):
+                long_term_memory_messages.append(msg)
+            elif hasattr(msg, "model_dump"):
+                long_term_memory_messages.append(msg.model_dump())
+
+        # For broadcast-triggered workflows, also include additional context memory
+        source = instance_data.get("source") if instance_data else None
```

Comment:
let's add a todo/issue to design getting context for agents/sub-agents, context engineering is important to keep on task, too much context and it can derail the agent